### PR TITLE
log: add customized logger for daemon and bpm (#2902)

### PR DESCRIPTION
**Pull request by STRRL**
### What problem does this PR solve?

- it is a part of https://github.com/chaos-mesh/chaos-mesh/issues/2969

### What's changed and how it works?

- cherry-pick #2902 into branch `release-2.1` for updating logger usage in chaos-daemon and bpm

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

Tests

- [x] Unit test
- [ ] E2E test
- [ ] No code
- [x] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by STRRL** [comment]
PTAL

/cc @Andrewmatilde @iguoyr

File: pkg/bpm/bpm.go [hunk 1]
```diff
@@ -409,6 +412,13 @@ func (b *ProcessBuilder) SetStderr(stderr io.ReadWriteCloser) *ProcessBuilder {
 	return b
 }
 
+func (b *ProcessBuilder) getLoggerFromContext(ctx context.Context) logr.Logger {
+	// this logger is inherited from the global one
+	// TODO: replace it with a specific logger by passing in one or creating a new one
+	logger := log.L().WithName("background-process-manager.process-builder")
```
[the remark this answers is not in the document]
Comment (STRRL) [reply pkg/bpm/bpm.go:420]: @sonatype-lift ignore

**Comment by iguoyr** [comment]
@STRRL Do we need to update `CHANGELOG.md`?

**Comment by STRRL** [comment]
> @STRRL Do we need to update `CHANGELOG.md`?

Yes. I would create PRs for updating CHANGELOG for both this PR and #2902

**Comment by STRRL** [comment]
> > @STRRL Do we need to update `CHANGELOG.md`?
> 
> Yes. I would create PRs for updating CHANGELOG for both this PR and #2902

Updated! PTAL. Another related PR: https://github.com/chaos-mesh/chaos-mesh/pull/3000

**Comment by iguoyr** [comment]
@STRRL Hi, I think files under `build/__pycache` should be added to `.gitignore`.

**Comment by STRRL** [comment]
> @STRRL Hi, I think files under `build/__pycache__` should be added to `.gitignore`.

Oops!  I have updated with git rebase, PTAL.

File: pkg/log/doc.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2022 Chaos Mesh Authors.
```
Comment (Andrewmatilde) [inline pkg/log/doc.go:1]: I think Readme is better than a go file

**Review by Andrewmatilde — COMMENTED** [review]
Rest LGTM.

File: pkg/log/doc.go (hunk 2 above)
Comment (STRRL) [reply pkg/log/doc.go:1]: I prefer to use godoc just nearby the code and follow the https://go.dev/doc/effective_go#commentary.

Using README might give a better experience on GitHub and local IDE, but that's the third-party tools' feature, not the language itself.

File: pkg/chaosdaemon/util.go [hunk 3]
```diff
@@ -74,7 +76,8 @@ func GetChildProcesses(ppid uint32) ([]uint32, error) {
 
 				reader, err := os.Open(statusPath)
```
[the remark this answers is not in the document]
Comment (STRRL) [reply pkg/chaosdaemon/util.go:77]: @sonatype-lift ignore

File: pkg/chaosdaemon/util.go [hunk 4]
```diff
@@ -74,7 +76,8 @@ func GetChildProcesses(ppid uint32) ([]uint32, error) {
...
 				if err != nil {
-					log.Error(err, "read status file error", "path", statusPath)
+					// TODO: replace this usage of global logger
+					log.L().WithName("chaos-daemon.util.GetChildProcesses").Error(err, "read status file error", "path", statusPath)
 					return
 				}
 				defer reader.Close()
```
[the remark this answers is not in the document]
Comment (STRRL) [reply pkg/chaosdaemon/util.go:83]: @sonatype-lift ignore

**Review by iguoyr — APPROVED** [review]
LGTM

File: pkg/chaosdaemon/server.go [hunk 5]
```diff
@@ -164,30 +193,30 @@ func StartServer(conf *Config, reg RegisterGatherer) error {
...
 		return err
 	}
 
 	g.Go(func() error {
-		log.Info("Starting http endpoint", "address", httpBindAddr)
+		logger.Info("Starting http endpoint", "address", httpBindAddr)
 		if err := httpServer.ListenAndServe(); err != nil {
-			log.Error(err, "failed to start http endpoint")
+			logger.Error(err, "failed to start http endpoint")
 			httpServer.Shutdown(context.Background())
```
[the remark this answers is not in the document]
Comment (STRRL) [reply pkg/chaosdaemon/server.go:210]: @sonatype-life ignore

File: pkg/bpm/bpm.go [hunk 6]
```diff
@@ -329,7 +331,8 @@ type ProcessBuilder struct {
 	stdout     io.ReadWriteCloser
 	stderr     io.ReadWriteCloser
 
-	ctx context.Context
+	ctx    context.Context
+	logger logr.Logger
```
[the remark this answers is not in the document]
Comment (STRRL) [reply pkg/bpm/bpm.go]: @sonatype-life ignore

[the remark this answers is not in the document]
Comment (STRRL) [reply pkg/bpm/bpm.go]: @sonatype-lift ignore

**Pull request merged**